holopy/scattering/tests/test_calculations.py

```diff
@@ -20,75 +20,178 @@
+                               [-2.41362056+1.74943249j, -2.15238106+2.50562808j]],
+                              [[ 0.        +0.j        ,  0.        +0.j        ],
+                               [ 0.        +0.j        ,  0.        +0.j        ]]],
+                             [[[ 0.        +0.j        ,  0.        +0.j        ],
+                               [ 0.        +0.j        ,  0.        +0.j        ]],
+                              [[-2.55065057+1.60766597j, -2.74726197+2.19673594j],
+                               [-2.74726197+2.19673594j, -2.66462981+2.81492861j]]]])
+        self.assertTrue(np.allclose(result.values.squeeze(), expected))
 
-class TestDetermineTheory(unittest.TestCase):
-    @attr("fast")
-    def test_determine_default_theory_for_on_sphere(self):
+    @attr('fast')
+    def test_finalize(self):
+        result = finalize(LOCATIONS.values, LOCATIONS)
+        expected = copy_metadata(LOCATIONS.values, LOCATIONS)
+        self.assertTrue(result.equals(expected))
+
+    @attr('medium')
+    def test_scattered_field_to_hologram(self):
+        size = 3
+        coords = np.linspace(0, 1, size)
+        scat = xr.DataArray(np.array([1, 0, 0]), coords=[('vector', coords)])
+        ref = xr.DataArray(np.array([1, 0, 0]), coords=[('vector', coords)])
+        normals = np.array((0, 0, 1))
+        correct_holo = (np.abs(scat + ref)**2).sum(dim='vector')
+        holo = scattered_field_to_hologram(scat, ref, normals)
+        self.assertEqual(holo.values.mean(), correct_holo.values.mean())
+
+
+class TestDetermineDefaultTheoryFor(unittest.TestCase):
+    @attr('fast')
+    def test_determine_default_theory_for_sphere(self):
         default_theory = determine_default_theory_for(Sphere())
         correct_theory = Mie()
         self.assertTrue(default_theory == correct_theory)
 
     @attr('fast')
-    def test_determine_default_theory_for_on_spheres(self):
+    def test_determine_default_theory_for_spheres(self):
         default_theory = determine_default_theory_for(
             Spheres([Sphere(), Sphere()]))
         correct_theory = Multisphere()
         self.assertTrue(default_theory == correct_theory)
 
     @attr('fast')
-    def test_determine_default_theory_for_on_spheroid(self):
+    def test_determine_default_theory_for_spheroid(self):
         scatterer = Spheroid(n=1.33, r=(1.0, 2.0))
         default_theory = determine_default_theory_for(scatterer)
         correct_theory = Tmatrix()
         self.assertTrue(default_theory == correct_theory)
 
     @attr('fast')
-    def test_determine_default_theory_for_on_cylinder(self):
+    def test_determine_default_theory_for_cylinder(self):
         scatterer = Cylinder(n=1.33, h=2, d=1)
         default_theory = determine_default_theory_for(scatterer)
         correct_theory = Tmatrix()
         self.assertTrue(default_theory == correct_theory)
 
+    @attr('fast')
+    def test_determine_default_theory_for_layered_sphere(self):
```

Comment:
I was actually suggesting to add a check of a `Spheres` object containing a layered sphere. Usually `Spheres` defaults to Multisphere theory, but that can't handle layered spheres, so there's a check in the `determine_default_theory_for` that looks for layered spheres and uses Mie superposition instead of multisphere if there are any present.